src/Staircase.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App;
+
+class Staircase
+{
+    public function print(int $size)
+    {
+        $staircase = [];
```

Comment:
Se `$staircase` fosse uma string, você conseguiria evitar a alocação de arrays e a utilização da função `implode`.

Caso prefira manter como string, recomendo fazer algo como `$staircase = array_fill(0, $size, null);`. Desta forma você pré-aloca um array já com `$size` posições. Isto vai te economizar alguns `malloc()` no meio do caminho.

Referências:
- [array_fill()](https://www.php.net/manual/en/function.array-fill)
- [Sobre pré alocar o array como tamanho correto sempre que possível (16:42 ~ 17:55)](https://youtu.be/JBWgvUrb-q8?t=1000)

Reply:
Preferi utilizar a string.
A dúvida ficou na utilização da quebra de linha (PHP_EOL): é melhor fazer um `rtrim()` no final para remover a última quebra de linha que é desnecessária ou utilizar um `if statement` para validar caso seja a última iteração não adicionar uma quebra de linha?